src/Grids/grid_generation.jl

```diff
@@ -150,3 +150,48 @@ function generate_coordinate(FT, ::Flat, N, H, ::Nothing, coordinate_name, arch)
         return FT(1), nothing, nothing, FT(1), FT(1)
     end
 end    
+
+#####
+##### ZStarVerticalCoordinate
+#####
+
+generate_coordinate(FT, ::Periodic, N, H, ::ZStarVerticalCoordinate, coordinate_name, arch, args...) = 
+    throw(ArgumentError("Periodic domains are not supported for ZStarVerticalCoordinate"))
+
+# Generate a moving coordinate with evolving scaling (`σ`) for spacings and znodes
```

Comment:
I don't think this function does what the comment says. A moving coordinate is presumably implemented by a model. The comment should be updated to state exactly what this function does. PResumably, it makes it _possible_ to have a moving coordinate, somehow.